spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/DefaultTokenServices.java

```diff
@@ -137,6 +139,11 @@ public OAuth2AccessToken refreshAccessToken(String refreshTokenValue, TokenReque
 			throw new InvalidGrantException("Invalid refresh token: " + refreshTokenValue);
 		}
 
+		if (refreshTokenValue == null) {
```

Comment:
The `refresh_token` validation should happen before it reaches `DefaultTokenServices`. We should move it to either `TokenEndpoint` or `RefreshTokenGranter`.